libcst/codemod/commands/tests/test_rename.py

```diff
@@ -0,0 +1,385 @@
+        self.assertCodemod(
+            before, after, old_name="a.z", new_name="d.z",
+        )
+
+    def test_other_import_froms_untouched(self) -> None:
+        before = """
+            from a import b, c, d
+
+            class Foo(b):
+                bar: c.bar
+                baz: d.baz
+        """
+        after = """
+            from a import c, d
+            from f import b
+
+            class Foo(b):
+                bar: c.bar
+                baz: d.baz
+        """
+        self.assertCodemod(
+            before, after, old_name="a.b", new_name="f.b",
+        )
+
+    def test_no_removal_of_import_in_use(self) -> None:
+        before = """
+            import a
+
+            class Foo(a.b):
+                pass
+            class Foo2(a.c):
+                pass
+        """
+        after = """
+            import a
+            import z
+
+            class Foo(z.b):
+                pass
+            class Foo2(a.c):
+                pass
+        """
+        self.assertCodemod(
+            before, after, old_name="a.b", new_name="z.b",
+        )
+
+    def test_no_removal_of_import_from_in_use(self) -> None:
+        before = """
+            from a import b
+
+            class Foo(b.some_class):
+                bar: b.some_other_class
+        """
+        after = """
+            from a import b
+            import blah
+
+            class Foo(blah.some_class):
+                bar: b.some_other_class
+        """
+        self.assertCodemod(
+            before, after, old_name="a.b.some_class", new_name="blah.some_class",
+        )
+
+    def test_other_unused_imports_untouched(self) -> None:
+        before = """
+            import a
+            import b
+
+            class Foo(a.obj):
+                pass
+        """
+        after = """
+            import b
+            import c
+
+            class Foo(c.obj):
+                pass
+        """
+        self.assertCodemod(
+            before, after, old_name="a.obj", new_name="c.obj",
+        )
+
+    def test_complex_module_rename(self) -> None:
+        before = """
+            from a.b.c import d
+
+            class Foo(d.e.f):
+                pass
+        """
+        after = """
+            import g.h.i
+
+            class Foo(g.h.i.j):
+                pass
+        """
+        self.assertCodemod(before, after, old_name="a.b.c.d.e.f", new_name="g.h.i.j")
+
+    def test_names_with_repeated_substrings(self) -> None:
+        before = """
+            from aa import aaaa
+
+            class Foo(aaaa.Bar):
+                pass
+        """
+        after = """
+            from b import c
+
+            class Foo(c.Bar):
+                pass
+        """
+        self.assertCodemod(
+            before, after, old_name="aa.aaaa", new_name="b.c",
+        )
+
+    def test_import_star_attr(self) -> None:
+        before = """
+            from foo import *
+
+            def baz():
+                foo.bar(5)
+        """
```

Comment:
assuming the fully-qualified name of the thing is `foo.bar`, this initial code is wrong, if you do `from foo import *` you'd have just `bar` in your namespace, not `foo.bar`.

I think this codemod should basically ignore `*` imports. So this test would be correct if we have an unrelated `*` import and then an `import foo`, but we should never make any assumptions about names coming from `*` imports.